docs/src/rules/no-inner-declarations.md

```diff
@@ -61,14 +61,15 @@ function doSomething() {
 
 ## Rule Details
 
-This rule requires that function declarations and, optionally, variable declarations be in the root of a program, or in the root of the body of a function, or in the root of the body of a class static block.
+This rule requires that function declarations and, optionally, variable declarations be in the root of a program, or in the root of the body of a function, or in the root of the body of a class static block in the non-strict code. This rule doesn't report function declarations and variable declarations if code is in strict mode (code with `"use strict"` tag).
 
 ## Options
 
-This rule has a string option:
+This rule has a string and an object option:
 
 * `"functions"` (default) disallows `function` declarations in nested blocks
 * `"both"` disallows `function` and `var` declarations in nested blocks
+* `{ legacy: false }` (default) if this option set to `true` then the rule will report function and variable declarations (according to the string options) only if the eslint is configured to `ecmaVersion: 5`.
```

Comment:
I don't think this option should apply to variable declarations, only function declarations.